### PR TITLE
[WIP][transaction][coordinator]Add persistent meta data store for transaction coordinator.

### PR DESCRIPTION
**Pull request by MarvinCai**
Add persistent meta data store backed by RocksDB.
Using KeyValueStorageRocksDB from Bookkeeper as RocksDB accessor. 

Pending doc and UTs and mechanism to config RocksDB.

### CODE REVIEW

File: pulsar-transaction/coordinator/pom.xml [hunk 1]
```diff
@@ -40,6 +40,11 @@
             <artifactId>pulsar-transaction-common</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.pulsar</groupId>
+            <artifactId>managed-ledger</artifactId>
+            <version>2.3.0-SNAPSHOT</version>
```
Comment (sijie) [inline pulsar-transaction/coordinator/pom.xml]: use `${project.version}`

**Comment by MarvinCai** [comment]
Thanks for the advice.
Yes, I also just realized, Coordinator need to be replicated so it can recover from single node failure.
But RockDB is only embedded KV store and is not replicated so only using RocksDB is not enough.
I'll go though the Table Service doc and see how we can utilize Statelib here: https://docs.google.com/document/d/155xAwWv5IdOitHh1NVMEwCMGgB28M3FyMiQSxEpjE-Y/edit#heading=h.56rbh52koe3f.

**Comment by sijie** [comment]
@MarvinCai is this pull request ready for review?

**Comment by MarvinCai** [comment]
@sijie I updated to use KVStore and add necessary configs for that, haven't add unit test yet. Can you take a quick look to see if I'm on the right track before I adding test for it? Thanks!

**Comment by sijie** [comment]
> Can you take a quick look to see if I'm on the right track before I adding test for it? Thanks!

The overall looks good to me.

**Pull request closed without merging**